src/Silex/CallbackResolver.php

```diff
@@ -33,40 +33,59 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
```

Comment:
I'm not sure when isValid() is called, does this means services have to be created before they can be used ?
why not keep just is_string() here and rely on convertCallback to throw exception when needed?